cmd/juju/application/refresher/refresher.go

```diff
@@ -169,7 +165,11 @@ func (d *localCharmRefresher) Allowed(_ RefresherConfig) (bool, error) {
 // Refresh a given local charm.
 // Bundles are not supported as there is no physical representation in Juju.
 func (d *localCharmRefresher) Refresh() (*CharmID, error) {
-	ch, newURL, err := d.charmRepo.NewCharmAtPathForceBase(d.charmRef, d.deployedBase, d.forceBase)
+	deployedBase, err := corebase.ParseBase(d.charmOrigin.Platform.OS, d.charmOrigin.Platform.Channel)
```

Comment:
Should this be "deployed" as in what is already there, vs just 'base'?
I'm not sure what is being parsed here, but it sure looks like you're parsing the charm you're about to deploy.

Doesn't need touching for this PR because you are leaving it with the name that it was using.
But I do want to be clear to the other humans reading this code whether this is the thing that is already deployed, or the thing that is about to become deployed (baseToDeploy)

Reply:
This is the operating system the charm has already been deployed with and we are looking for a newer charm supporting the same operating system